File: MATEMÁTICA/1555/main.dart
import 'dart:io';
import 'dart:math';

String winner(int x, int y) {
  num rafael = pow(3 * x, 2) + pow(y, 2);
  num beto = 2 * pow(x, 2) + pow(5 * y, 2);
  num carlos = -100 * x + pow(y, 3);
  
  if (rafael > beto && rafael > carlos)
    return 'Rafael';
  else if (beto > rafael && beto > carlos)
    return 'Beto';
  return 'Carlos';
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  for (int times = 0; times < n; times++) {
    var [x, y] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    print('${winner(x, y)} ganhou');
  }
}

File: MATEMÁTICA/1555/main.rb
# frozen_string_literal:true

def winner(x, y)
  rafael = (3 * x)**2 + y**2
  beto = 2 * x**2 + (5 * y)**2
  carlos = -100 * x + y**3

  return 'Rafael' if rafael > beto && rafael > carlos
  return 'Beto' if beto > rafael && beto > carlos

  'Carlos'
end

n = gets.to_i
n.times do
  x, y = gets.split.map(&:to_i)
  puts "#{winner(x, y)} ganhou"
end
